### PR TITLE
Sddc hosts IT  coverage

### PR DESCRIPTION
**Pull request by jain-prerna**

### CODE REVIEW

**Comment by jain-prerna** [comment]
This PR is dependent on https://github.com/saltstack/salt-ext-modules-vmware/pull/251 for a fixture in conftest.py that i have added there.

- sddc hosts modules are already merged, this PR  has IT coverage and some naming improvisation in the module

- the maintenance use case(when creation/deletion is taking long time to execute) is different from sddc_cluster, So i have handled slightly different here

**Review by waynew — COMMENTED** [review]
This looks mostly good (similar question to #251 about the config/common data fixture) :thinking: 

Also still need to be able to run this locally :upside_down_face:

**Comment by jain-prerna** [comment]
> This looks mostly good (similar question to #251 about the config/common data fixture) 🤔
> 
> Also still need to be able to run this locally 🙃

**Comment by jain-prerna** [comment]
> This looks mostly good (similar question to #251 about the config/common data fixture) 🤔
> 
> Also still need to be able to run this locally 🙃

Hi @waynew , i have modified the vmc_connect fixture and common_data  fixtures

File: tests/integration/modules/test_vmc_sddc_host.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    )
+    response.raise_for_status()
+    return response.json()
+
+
+@pytest.fixture
+def get_primary_cluster_id(common_data, salt_call_cli):
+    ret = salt_call_cli.run("vmc_sddc_clusters.get_primary", **common_data)
+    result_as_json = ret.json
+    return result_as_json["cluster_id"]
```
Comment (waynew) [inline tests/integration/modules/test_vmc_sddc_host.py:48]: **suggestion** this returned None for me, using the SALT-SDDC-TEST-NEW SDDC. If I should be using a different one for testing, please provide the name

**Review by waynew — CHANGES_REQUESTED** [review]
Running tests against the SALT-SDDC-TEST-NEW unfortunately did not work:
```
    @pytest.fixture
    def get_primary_cluster_id(common_data, salt_call_cli):
        ret = salt_call_cli.run("vmc_sddc_clusters.get_primary", **common_data)
        result_as_json = ret.json
>       return result_as_json["cluster_id"]
E       TypeError: 'NoneType' object is not subscriptable
```

File: tests/integration/modules/test_vmc_sddc_host.py (hunk 1 above)
Comment (jain-prerna) [reply tests/integration/modules/test_vmc_sddc_host.py:48]: @waynew, You have to use "Prerna-Multi-host-SDDC" 
SDDC ID - d088f189-d3a3-43de-92b3-6c4ac26f9ac2
this is the same one i shared with you on slack channel.

**Comment by shaikres** [comment]
This PR is dependent on [PR-251](https://github.com/saltstack/salt-ext-modules-vmware/pull/251)
After the PR-251 is merged, rebase this PR and run the tests

File: tests/integration/modules/test_vmc_sddc_host.py (hunk 1 above)
Comment (shaikres) [reply tests/integration/modules/test_vmc_sddc_host.py:48]: @waynew, please use new SDDC -  Salt-Multihost-SDDC-reshma (e5e387a1-216b-4002-a549-59b24126af5a) to run the tests.

**Comment by shaikres** [comment]
Hi @waynew,
I rebased this PR with latest changes from main branch. Now this is ready for review.

**Review by waynew — APPROVED** [review]
Tests are passing on this one, too

**Pull request merged**